src/native/corehost/apphost/static/CMakeLists.txt

```diff
@@ -14,14 +14,14 @@ if (NOT CLR_CMAKE_TARGET_OSX)
     set(CMAKE_INSTALL_RPATH "\$ORIGIN/netcoredeps")
 endif()
 
-set(SKIP_VERSIONING 1)
-
 include_directories(..)
 include_directories(../../json)
 
 include_directories(${CLR_SRC_LIBS_NATIVE_DIR}/AnyOS/zlib)
 if(NOT CLR_CMAKE_TARGET_WIN32)
     include_directories(${CLR_SRC_LIBS_NATIVE_DIR}/Unix/Common)
+else()
+    set(SKIP_VERSIONING 1)
```

Comment:
Don't we have the same problem on Windows too?

Reply:
@janvorli, I don't have Windows build system at the moment. I spent some time figuring out the error on CI machine.
The native.rc file in corehost requires version_info.h, which we don't generate for singlefilehost (now that it builds from coreclr partition). `fatal error C1083: Cannot open include file: 'version_info.h'`

coreclr, libraries and mono all use the file name _version.h. corehost calls it version_info.h. I aligned the name, and passed the cmake path variable from coreclr to corehost to use it in `include_directories` for singlefilehost like this: https://github.com/dotnet/runtime/commit/1190c6dcf8556174f2ba0a05f2643705d8aaf13a. That fixes the error C1083.

Then I was getting these: `error RC2104 : undefined keyword or key name: VS_FFI_FILEFLAGSMASK`. I fixed this one by definition it explicitly (as you can see in that commit). Then the next error was `error RC2104 : undefined keyword or key name: VS_FF_DEBUG`, where I stopped pursuing  and confined the fix for Unix-only.

If you have an idea how to avoid these definitions for rc, please share, I will give it a try. 🙂

Reply:
We should NOT set the version resource on Windows - that's for the app to set, not for us. I think it would be good to include the version info somewhere in the binary though - so that it's possible to "grep" for it if necessary.

Reply:
Yes, I read the conversation. So we are aligned here. I have simplified the patch so we emit info in all three hosts (dotnet, apphost and singlefilehost) on Unix.

> I think it would be good to include the version info somewhere in the binary though

Agreed, but we never generated version.c on Windows, so it would require a separate effort involving changes in arcade. see https://github.com/dotnet/arcade/blob/e7ede87875f41a9b3df898ae08da5ebc96e24f56/src/Microsoft.DotNet.Arcade.Sdk/tools/Version.targets#L113.

Reply:
Makes sense.